src/VecSim/spaces/IP/IP_AVX512_FP32.cpp

```diff
@@ -18,7 +18,6 @@ float FP32_InnerProductSIMD16Ext_AVX512_impl(const void *pVect1v, const void *pV
         __m512 v2 = _mm512_loadu_ps(pVect2);
         pVect2 += 16;
         sum512 = _mm512_add_ps(sum512, _mm512_mul_ps(v1, v2));
-        // sum512 = _mm512_fmadd_ps(v1, v2, sum512);
     }
 
     return _mm512_reduce_add_ps(sum512);
```

Comment:
what does this do? arent there similar function to reduce to float or double from 128 (instead of the summation from Tmp we do in all other functions)

Reply:
It reduces the 512bit register into a single float scalar by summing its vectorized floats. I Couldn't  find an equivalent function for 256 or 128 registers...